[#c0eutiqt]/1028 fix discreetly

minhducvt293 opened the pull request:
* **Trello Ticket**: https://trello.com/c/cOeutIqt/1028-fix-discreetly
* Other Relevant Links [google-cloud-kms 2.0.0 migration guide](https://googleapis.dev/python/cloudkms/latest/UPGRADING.html)

# What does this Pull Request do?
- update discreetly to use google-cloud-kms version >=2.0.0 and <3.0.0
- update tests for test-gcp accordingly
- update docs for new changes
- update .envrc for python 3.7.3
- add template for making commit and PRs

prepare to release `discreetly` v0.2.0

# How should this be validated?
1. uninstall `google-cloud-kms`
```console
pip uninstall google-cloud-kms
```

2. install dependencies:
```console
pip install -r requirements.txt
```

3. run pytest should be sufficient
```console
pytest
```

4. Follow the `README.md` and verify those functions work correctly.

# Checklist
<!-- This is a reminder checklist that you've taken all necessary steps to prepare this PR
Replace `[ ]` with `[x]` for all that apply.-->
- [x] You've fetched and rebased your branch on an **up-to-date** version `master`
- [x] PR contains new tests relevant for the feature/problem this PR is addressing
- [x] All tests and linting pass
- [x] Your commits have been squashed/rebased/etc. for reviewer clarity


# Additional Notes:
This PR is for release of v0.2.0 which supports `python` 3 only
Other PR (https://github.com/tra-sg/discreetly/pull/11) is for release of v0.1.2 which also supports `python` 2

# Interested parties
@ghosalya as main reviewer
@eddies

ghosalya commented on .envrc:
```diff
@@ -4,7 +4,7 @@
 #   see: https://github.com/direnv/direnv/wiki/Python#pyenv
 if has pyenv; then
-  use pyenv 2.7.16 3.7.3
+  use pyenv 3.7.3
```
I was not able to run `use pyenv`, I think this recently changes to `layout pyenv` by direnv. IIRC `use pyenv` was something we needed to declare ourselves, since previously pyenv was not supported out-of-the-box.

ghosalya commented on .github/workflows/tests.yml:
```diff
@@ -6,25 +6,29 @@ jobs:
             ${{ runner.os }}-pip-
       - name: Install dependencies
-        if: steps.cache.outputs.cache-hit != 'true'
+        # if: steps.cache.outputs.cache-hit != 'true'
```
why is this commented out? It wasn't commented out in the other PR

ghosalya commented on tox.ini:
```diff
@@ -11,8 +11,8 @@
 [tox]
-# envlist = py{27,35,36,37,38}
-envlist = py{27,37}
+# envlist = py{36,37,38}
```
any reason why we only have 1 env instead of 3?

minhducvt293 committed:
[#cOeutIqt] 1028 fix discreetly
- update discreetly to use google-cloud-kms version >=2.0.0 and <3.0.0
- update tests for test-gcp accordingly
- update docs for new changes
- update .envrc for python 3.7.3

ghosalya reviewed (approved):
lgtm

note: we are expecting a release pr for 0.2.0 after this PR
